ruma-events/src/room/message.rs

```diff
@@ -85,6 +85,337 @@ impl MessageEventContent {
+
+        let notice_message_content = NoticeMessageEventContent { body, formatted: None };
+
+        Self {
+            msgtype: MessageType::Notice(notice_message_content),
+            relates_to: Some(Relation::Reply {
+                in_reply_to: InReplyTo { event_id: original_message.event_id.clone() },
+            }),
+            #[cfg(feature = "unstable-pre-spec")]
+            new_content: None,
+        }
+    }
+
+    /// A constructor to create a html text notice reply to a message.
+    pub fn notice_reply_html(
+        reply: impl Into<String>,
+        html_reply: impl Into<String>,
+        original_message: &MessageEvent,
+    ) -> Self {
+        let quoted = MessageEventContent::get_plain_quote_fallback(original_message);
+        let quoted_html = MessageEventContent::get_html_quote_fallback(original_message);
+
+        let body = format!("{}\n\n{}", quoted, reply.into());
+        let formatted = format!("{}\n\n{}", quoted_html, html_reply.into());
+
+        let notice_message_content =
+            NoticeMessageEventContent { body, formatted: Some(FormattedBody::html(formatted)) };
+
+        Self {
+            msgtype: MessageType::Notice(notice_message_content),
+            relates_to: Some(Relation::Reply {
+                in_reply_to: InReplyTo { event_id: original_message.event_id.clone() },
+            }),
+            #[cfg(feature = "unstable-pre-spec")]
+            new_content: None,
+        }
+    }
+
+    fn get_plain_quote_fallback(original_message: &MessageEvent) -> String {
+        match &original_message.content.msgtype {
+            MessageType::Audio(_) => {
+                format!("> <{:?}> sent an audio file.", original_message.sender)
+            }
+            MessageType::Emote(content) => {
+                format!("> * <{:?}> {}", original_message.sender, content.body)
+            }
+            MessageType::File(_) => {
+                format!("> <{:?}> sent a file.", original_message.sender)
+            }
+            MessageType::Image(_) => {
+                format!("> <{:?}> sent an image.", original_message.sender)
+            }
+            MessageType::Location(content) => {
+                format!("> <{:?}> {}", original_message.sender, content.body)
+            }
+            MessageType::Notice(content) => {
+                format!("> <{:?}> {}", original_message.sender, content.body)
+            }
+            MessageType::ServerNotice(content) => {
+                format!("> <{:?}> {}", original_message.sender, content.body)
+            }
+            MessageType::Text(content) => {
+                format!("> <{:?}> {}", original_message.sender, content.body)
+            }
+            MessageType::Video(_) => {
+                format!("> <{:?}> sent a video.", original_message.sender)
+            }
+            MessageType::_Custom(content) => {
+                // Eagerly deserialize `body` since we assume every message should have it
```

Comment:
There is no deserialization going on here, just accessing an entry in a map.

Reply:
Yes, I had misread an earlier comment. Removed the comment.